src/plugins/mf_priority.cpp

```diff
@@ -60,25 +62,51 @@ std::map<std::string, int> priority_weights;
  *
  * queue: a factor that can further affect the priority of a job based on the
  *     queue passed in.
+ *
+ * age: a factor that considers the time that a job was released to the
+ *     scheduler to be run but could not due to another constraint (e.g a
+ *     resource constraint). Any urgency, priority, or jobspec-update event
+ *     posted to a job in SCHED state will transition the job back to
+ *     PRIORITY. After the priority of the job is calculated, the job will
+ *     be transitioned back to SCHED state and the job's "t_released"
+ *     timestamp will be updated to reflect this most recent transition. This
+ *     is to prevent users from purposely holding their jobs while they are in
+ *     SCHED state and artificially boosting their job's priority.
  */
 int64_t priority_calculation (flux_plugin_t *p,
                               flux_plugin_arg_t *args,
                               int userid,
                               char *bank,
                               int urgency)
 {
+    long int jobid;
     double fshare_factor = 0.0, priority = 0.0;
     int queue_factor = 0;
-    int fshare_weight, queue_weight;
+    int fshare_weight, queue_weight, age_weight;
+    double t_released = 0.0, t_current = 0.0, age_factor = 0.0;
     Association *b;
 
+    flux_t *h = flux_jobtap_get_flux (p);
+    if (flux_plugin_arg_unpack (args,
+                                FLUX_PLUGIN_ARG_IN,
+                                "{s:I}",
+                                "id", &jobid) < 0) {
+        flux_log (h,
+                  LOG_ERR,
+                  "flux_plugin_arg_unpack: %s",
+                  flux_plugin_arg_strerror (args));
+        return -1;
+    }
+
     fshare_weight = priority_weights["fairshare"];
     queue_weight = priority_weights["queue"];
+    age_weight = priority_weights["age"];
 
     // check values of priority factor weights; if not configured,
     // these will be set to -1, so just use default weights
     if (fshare_weight == -1) fshare_weight = 100000;
     if (queue_weight == -1) queue_weight = 10000;
+    if (age_weight == -1) age_weight = 1000;
```

Comment:
Cleanup suggestion: place defaults as defines at top of the file and initialize each `priority_weights[factor] = DEFAULT`` so that you don't have to repeat these checks for every job.